src/main/java/io/reactivex/plugins/RxJavaPlugins.java

```diff
@@ -424,9 +436,28 @@ static boolean isBug(Throwable error) {
     }
 
     static void uncaught(@NonNull Throwable error) {
-        Thread currentThread = Thread.currentThread();
-        UncaughtExceptionHandler handler = currentThread.getUncaughtExceptionHandler();
-        handler.uncaughtException(currentThread, error);
+        ThrowablePermittingConsumer<? super Throwable> f = onUncaughtHandler;
+        if (f == null) {
+            Thread currentThread = Thread.currentThread();
+            UncaughtExceptionHandler handler = currentThread.getUncaughtExceptionHandler();
+            handler.uncaughtException(currentThread, error);
+        } else {
+            try {
+                f.accept(error);
+            } catch (Throwable e) {
+                RxJavaPlugins.<Error>sneakyThrow(e);
```

Comment:
I'd deliver to current thread uncaught exception handler similarly how this method worked before

This change can also end up violating Reactive Streams in some cases

Reply:
The whole point of this is to avoid doing that though, but we have to try/catch this per the signature of the accept() method. I can avoid this by removing the signature, but not sure if that'd have any tradeoffs for consumers that just want to immediately rethrow.

How would it violate them? Note that this change is not the default behavior, which remains unchanged.

Reply:
The goal is to enable crashing though, as the current behavior of just handing it to the nearest uncaught exception handler is what this wants to facilitate changing.

What about taking artem's suggestion and making this a custom hook to just provide the uncaughtexceptionhandler? I.e. - no consumer or anything, just allow specifying a custom one and fall back to the current behavior of fetching it out of the current thread?

Reply:
You are still trying to backdoor a backdoor. The overall issue sounds like you want to hook your developers who throw in their `onError` handler to get away with failures. What makes you think they wouldn't do it for this extra hook?

Reply:
onError will still funnel thrown exceptions through the current uncaught() implementation since it catches exceptions from any installed onError handler. It's not to get away with failures at all, but rather to

- in junit, actually fail tests when they have unhandled errors
- in Android - make them recoverable if this is just a delegating observern. The goal is to be able to try/catch without something calling system.exit() for us under the hood

Both require actually throwing the exception rather than silently passing it to an uncaught exception handler. I would be all for just using the one from handler API if it actually allowed for throwing exceptions that could be caught from outside the onError() method, but that's not possible at the moment

Reply:
The repeatedly printed stacktrace is not relevant if you want to fail the test.

The uncaught handlers are thread-local and you only need to get past the catch around the onError handler. They are invoked together on the same thread.

Reply:
Haven't forgotten about this, just doing some further testing. While this works in a test, on an actual android app it seems to get stuck in a loop. Trying to dig in more to understand why the runtime is behaving different than the test